### DISCUSSION TITLE
Essentials edition not activated after upgrade-restart

### DISCUSSION THREAD

**Discussion by augustr** [photoprism/photoprism#3813] [Q&A] [answered]
Hi!

I have installed PhotoPrism in a LXC Container under Proxmox. I build PhotoPrism from source and run it as root, following guides such as [this one](https://github.com/IndrekHaav/photoprism-debian#installing-photoprism-on-debian). I had major problems at first getting PhotoPrism to choose /var/lib/photoprism as storage-dir since it always tried <working-dir-or-home-dir>/.photoprism/storage. Startup params, env file or options.yml didn't help. I worked around that by modifying source.

The install seems to be working, but upgrading to essentials edition doesn't. It says the activation is successful and will take effect after restart. However, after restart the edition is still community. I've spent some hours debugging the code base but haven't found the root cause so far. I've tried both the release and the develop branch.

Any help appreciated. Thanks!

/August
Experienced coder and linux admin.

**Comment by lastzero** [comment]
I'll try to look into this tomorrow as today was extremely busy for us again. It's probably some kind of configuration issue, but it's hard to tell without spending more time on it.

**Reply by augustr** [reply]
Thanks. I can reiterate here the latest experiment that I wrote about in the chat:

Made the following change:

```
diff --git a/internal/hub/config.go b/internal/hub/config.go
index 70155f2b8..237209d2d 100644
--- a/internal/hub/config.go
+++ b/internal/hub/config.go
@@ -272,7 +272,9 @@ func (c *Config) ReSync(token string) (err error) {
        }
 
        // Decode JSON response.
+       fmt.Printf("Config Before: %+v\n", c)
        err = json.NewDecoder(r.Body).Decode(c)
+       fmt.Printf("Config After: %+v\n", c)
 
        if err != nil {
                return err
```

It gives (masked out some sensitive info):
```
Config Before: &{Version:231012-63f708417-Linux-AMD64 FileName:/var/lib/photoprism/hub.yml Key:6...b Secret:f...9 Session:0...a session:<nil> sessionMu:{state:0 sema:0} Status: Serial:z...i Env: UserAgent:PhotoPrism/231012-63f708417-Linux-AMD64 (debug; settings; z...i) PartnerID:}
```

```
Config After: &{Version:231012-63f708417-Linux-AMD64 FileName:/var/lib/photoprism/hub.yml Key:6...b Secret:f...9 Session:8...b session:<nil> sessionMu:{state:0 sema:0} Status: Serial:z...i Env: UserAgent:PhotoPrism/231012-63f708417-Linux-AMD64 (debug; settings; z...i) PartnerID:}
```

"Status: essentials" is not in the returned data from your web backend.

**Reply by lastzero** [reply]
You probably don't need to change anything there, because none of our members have problems with it. It could be a problem with the configuration of the backup path, for example, because there are some "is writable" checks not only for the main config folder. As I said, it is not possible to properly assess the problem without spending more time.

**If you could please email us the output of the `photoprism config` command to [membership@photoprism.app](https://www.photoprism.app/contact), that would be very helpful. Thanks a lot!**

**Reply by augustr** [reply]
I have now emailed the output of `photoprism config` to membership@photoprism.app . Thanks in advance.

**Reply by lastzero** [reply]
@augustr I would recommend having the config path set automatically or changing it to `/var/lib/photoprism/config` (also make sure it is writable even if you run this as root) and then try the activation again with a build of our latest code:

- What is definitely strange about your configuration is that the config path is the same as the storage path. Based on the output of the config command, both paths are `/var/lib/photoprism` right now.
- Instead, we recommend configuring only a storage path inside which the config path will then be automatically created on startup, so in your case `/var/lib/photoprism/config`.
- When you rebuild the binary using the latest code in our `develop` or `preview` branch, you will get a clearer warning message if the API keys could not be written to disk.

Should you then still have unexplainable issues, we will be happy to help if you email us at [membership@photoprism.app](https://www.photoprism.app/contact)!

**Comment by augustr** [comment, marked as answer]
For others reading this; this was solved. The root cause was not completely understood, but probably an initial first installation had some misconfiguration where registration and activation of "Essentials" worked partly. In later re-installations from scratch the problem persisted due to re-using the activation tokens. This had to be reset by PhotoPrism support on the upgrade server backend side.

Please correct me if I'm wrong, @lastzero.

I could try to reproduce the original problem (which seemed to have to do with PhotoPrism not picking up my environment variables) but I've spent enough time troubleshooting so I don't have more time for that.

Thanks for the support!
